### DISCUSSION TITLE
changing variable via setTimeout worked at second blur event

### DISCUSSION THREAD

**Discussion by lioutall** [nuejs/nue#370] [Q&A] [answered]
<input :if="show" @blur="hide"></input>

<script>
constructor(data) {
      this.show = true;
    }

hide() {
      setTimeout(() => {this.show = false}, 200);
  }
</script>

ubuntu 24
Nue 0.5.3 • Bun 1.1.8

**Comment by nobkd** [comment, marked as answer]
I think the timeout breaks the update (probably because the internal `update()` is called before the timeout is done), so try doing one of the following:

```js
hide() { setTimeout(() => this.update({show: false}), 200); }
// or
hide() { setTimeout(() => {this.show = false; this.update()}, 200); }
```
